abuild now builds with the '-Werror=format-security' option that leads to this
error:

  In file included from osdep.c:56:
  flocklnx.c: In function 'safe_flock':
  flocklnx.c:60:38: error: format not a string literal and no format arguments [-Werror=format-security]
     60 |       if (!logged++) syslog (LOG_ERR,tmp);
        |                                      ^~~

This is reproduced only on glibc, because it declares syslog() with the
'__format__' attribute:

  extern void syslog (int __pri, const char *__fmt, ...)
       __attribute__ ((__format__ (__printf__, 2, 3)));

--- a/src/osdep/unix/flocklnx.c
+++ b/src/osdep/unix/flocklnx.c
@@ -57,7 +57,7 @@
     case ENOLCK:		/* lock table is full */
       sprintf (tmp,"File locking failure: %s",strerror (errno));
       mm_log (tmp,WARN);	/* give the user a warning of what happened */
-      if (!logged++) syslog (LOG_ERR,tmp);
+      if (!logged++) syslog (LOG_ERR,"%s",tmp);
 				/* return failure if non-blocking lock */
       if (op & LOCK_NB) return -1;
       sleep (5);		/* slow down in case it loops */
